Add WPLV2 [WARPLABS]

enricomilan opened the pull request:
**Checklist** (✓/✕, or y/n)
- [x] passed schematics review
- [x] passed hardware samples testing
- [x] follows guidelines
- [x] follows connector standards
- [x] flight tested
- [x] comments/issues resolved


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added support for the WPLV2 flight controller (STM32H743) with SPI gyro and accelerometer and SPI flash storage.
  * Configured eight motor outputs, servo pins, beeper, LED strip (plus extra LED), multiple UARTs, I2C, ADC inputs (VBAT/current) and VTX enable.
  * OSD and LED strip enabled by default; default blackbox, current meter and voltage meter sources set.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

enricomilan committed:
new manufacturer WARPLABS and new config file WPLV2

haslinghuis commented on configs/WPLV2/config.h:
```diff
@@ -0,0 +1,114 @@
+#define USE_ACC_SPI_ICM42688P
+#define USE_FLASH
+#define USE_FLASH_W25Q128FV
+#ifndef USE_OSD_HD
+#define USE_OSD_HD
+#endif
+
+#define USE_LED_STRIP
+
```
LED_STRIP and OSD are user build options.
```suggestion
#define USE_BARO
```

haslinghuis commented on configs/WPLV2/config.h:
```diff
@@ -0,0 +1,114 @@
+
+#define DEFAULT_FEATURES (FEATURE_OSD | FEATURE_LED_STRIP)
+
+#define MSP_DISPLAYPORT_UART            SERIAL_PORT_USART2
```
```suggestion
#ifdef USE_OSD_HD
#define MSP_DISPLAYPORT_UART            SERIAL_PORT_USART2
#endif
```

enricomilan committed:
updated config with suggestions

osirisinferi commented on configs/WPLV2/config.h:
```diff
@@ -0,0 +1,114 @@
+
+#define ADC1_DMA_OPT   11
+#define ADC2_DMA_OPT   12
+#define ADC3_DMA_OPT   13
```
Oh and ADC3 is required for the internal voltage/temperature measurements, so that one needs to have DMA :slightly_smiling_face:

enricomilan committed:
fixed beeper, set pinio 1 config to non inverted

haslinghuis commented on configs/WPLV2/config.h:
```diff
@@ -0,0 +1,114 @@
+#define FLASH_SPI_INSTANCE              SPI2
+#define GYRO_1_SPI_INSTANCE             SPI1
+#define GYRO_2_SPI_INSTANCE             SPI1
+#define BARO_I2C_INSTANCE               I2CDEV_1
```
```suggestion
#define BARO_I2C_INSTANCE               I2CDEV_1
#define MAG_I2C_INSTANCE                I2CDEV_1
```

haslinghuis commented on configs/WPLV2/config.h:
```diff
@@ -0,0 +1,114 @@
+
+//#define USE_BARO
+//#define USE_BARO_DPS310
+
```
Remove when not used
```suggestion
```

enricomilan commented on configs/WPLV2/config.h:
```diff
@@ -0,0 +1,114 @@
+//#define USE_BARO
+//#define USE_BARO_DPS310
+
+#define USE_SERVOS
```
ok

enricomilan committed:
removed the barometer and followed the suggestions

osirisinferi commented on configs/WPLV2/config.h:
```diff
@@ -0,0 +1,109 @@
+#define DEFAULT_BLACKBOX_DEVICE         BLACKBOX_DEVICE_FLASH
+#define DEFAULT_CURRENT_METER_SOURCE    CURRENT_METER_ADC
+#define DEFAULT_VOLTAGE_METER_SOURCE    VOLTAGE_METER_ADC
+#define DEFAULT_CURRENT_METER_SCALE     100
```
Is this FC going to be part of a stack?

enricomilan committed:
Update configs/WPLV2/config.h

standard comment on file top, as suggested.

Co-authored-by: ot0tot <36753790+ot0tot@users.noreply.github.com>

enricomilan committed:
Update configs/WPLV2/config.h

ok, as suggested

Co-authored-by: Osiris Inferi <github@flut.nl.eu.org>

enricomilan committed:
added manufacturer id on manufacturers.md removed orphan gyro 2 spi def

fixing coderabbit suggestion

enricomilan committed:
Update configs/WPLV2/config.h

Co-authored-by: Osiris Inferi <github@flut.nl.eu.org>

osirisinferi reviewed (commented):
Just a last suggestion, otherwise looks good to me 🙂

enricomilan commented on configs/WPLV2/config.h:
```diff
@@ -0,0 +1,128 @@
+#define PINIO2_CONFIG 1
+
+#define DEFAULT_FEATURES (FEATURE_OSD | FEATURE_LED_STRIP)
+
```
It's good, the FC will have in the wiring diagram the suggestion for wiring RX on UART 1

enricomilan committed:
Update configs/WPLV2/config.h

It's good, the FC will have in the wiring diagram the suggestion for wiring RX on UART 1

Co-authored-by: Osiris Inferi <github@flut.nl.eu.org>

enricomilan committed:
Merge branch 'master' into warplabs

haslinghuis commented on configs/WPLV2/config.h:
```diff
@@ -0,0 +1,130 @@
+#define ADC_CURR_PIN PC2
+
+#define VTX_ENABLE_PIN PE3
+#define PINIO1_PIN VTX_ENABLE_PIN
+#define PINIO1_BOX 40
+#define PINIO1_CONFIG 129
+#define PINIO2_PIN PE4                  
+#define PINIO2_BOX 41
+#define PINIO2_CONFIG 1
```
Please add BOXNAMES to PINIO 1 and 2

enricomilan replied:
added BOXNAMES in commit 1d46088

enricomilan committed:
added box names as suggested

osirisinferi commented on configs/WPLV2/config.h:
```diff
@@ -0,0 +1,132 @@
+#define I2C1_SCL_PIN PB8
+#define I2C1_SDA_PIN PB9
+
+#define LED0_PIN PE2
```
The BF default is that the status LEDs are sinking their current 'into' the MCU, while in the board the MCU is sourcing the current. Not an issue, but requires the status LED to be inverted in software:

```suggestion
#define LED0_PIN PE2
#define LED0_INVERTED
```

enricomilan replied:
ok

enricomilan committed:
Update configs/WPLV2/config.h

reverse led logic to source current instead of sinking, as suggested

Co-authored-by: Osiris Inferi <github@flut.nl.eu.org>

osirisinferi reviewed (commented):
LGTM.

ot0tot commented on configs/WPLV2/config.h:
```diff
@@ -0,0 +1,133 @@
+
+#define I2C1_SCL_PIN PB8
+#define I2C1_SDA_PIN PB9
+
```
Should include these so externally attached baro/mag is detected.
```suggestion
#define MAG_I2C_INSTANCE I2CDEV_1
#define BARO_I2C_INSTANCE I2CDEV_1
```

enricomilan committed:
Update configs/WPLV2/config.h

Co-authored-by: ot0tot <36753790+ot0tot@users.noreply.github.com>

haslinghuis commented on configs/WPLV2/config.h:
```diff
@@ -0,0 +1,134 @@
+#define PINIO2_CONFIG 1
+#define BOX_USER2_NAME "AUX SW"
+
+#define DEFAULT_FEATURES (FEATURE_OSD | FEATURE_LED_STRIP)
```
These are user settings. There is no MAX7456

osirisinferi replied:
> There is no MAX7456

That's correct, the FC doesn't have an OSD IC.

haslinghuis replied:
```suggestion
```

haslinghuis commented on configs/WPLV2/config.h:
```diff
@@ -0,0 +1,134 @@
+    TIMER_PIN_MAP(  8, LED_STRIP_PIN, 1,  8 ) /* TIM2 CH1 */ \
+    TIMER_PIN_MAP( 9, SERVO1_PIN,    1, -1 ) /* TIM15 CH1 */ \
+    TIMER_PIN_MAP( 10, SERVO2_PIN,    1, -1 ) /* TIM15 CH2 */\
+    TIMER_PIN_MAP( 11, GYRO_1_CLKIN_PIN,    2, -1 ) /* TIM5 CH1 */
```
Looks wrong on github - fix indentation - and move timer comments outside the mapping

enricomilan replied:
fixed in commit 90c2ec009e12a98cccc46340602af1b96b7061ed

enricomilan committed:
Update configs/WPLV2/config.h

Co-authored-by: Mark Haslinghuis <mark@numloq.nl>

enricomilan committed:
fix indentation, moved timer comments outside the mapping
